training/run_main_ds.py

```diff
@@ -0,0 +1,182 @@
+import argparse
+import os
+import subprocess
+
+from instructlab.training import (
+    TorchrunArgs,
+    TrainingArgs,
+    DeepSpeedOptions,
+    config
+)
+
+from instructlab.training.utils import StreamablePopen
+
+
+def run_main_ds(torch_args: TorchrunArgs, train_args: TrainingArgs) -> None:
+    """
+    Wrapper around the main training job that calls torchrun.
+    """
+
+    if not os.path.exists(train_args.ckpt_output_dir):
+        os.makedirs(train_args.ckpt_output_dir, exist_ok=True)
+    command = [
+        "torchrun",
+        f"--nnodes={torch_args.nnodes}",
+        f"--node_rank={torch_args.node_rank}",
+        f"--nproc_per_node={torch_args.nproc_per_node}",
+        f"--rdzv_id={torch_args.rdzv_id}",
+        f"--rdzv_endpoint={torch_args.rdzv_endpoint}",
+        f"training/src/instructlab/training/main_ds.py",
```

Comment:
You probably want this to be its location in the container right? This will only work if you've cloned the repo which we are no longer doing.

I think `/usr/local/lib/python3.11/site-packages/instructlab/training/src/instructlab/training/main_ds.py` where you'd fine it.

Reply:
good catch :)